basil/utils/USBBinds.py

```diff
@@ -30,8 +30,11 @@ def query_identification(rm, resource, baud_rate, read_termination=None, write_t
     inst.baud_rate = baud_rate
     inst.read_termination = read_termination
     inst.write_termination = write_termination
-
-    return inst.query("*IDN?", delay=0.5)
+    try:
+        reply = inst.query("*IDN?", delay=0.1)
+    except pyvisa.VisaIOError:
+        reply = inst.query("*IDN?", delay=0.1)
+    return reply
 
```

Comment:
Why do you need a second try that is done in an `except` block? Does it always work on the second try and if not: will the whole script will crash? Or we add a loop with `x` tries that breaks when there is a reply?